// @syncfusion/ej2-vue-buttons/src/speed-dial/items.directive.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { EJComponentDecorator, allVue, gh, isExecute } from '@syncfusion/ej2-vue-base';

import { isNullOrUndefined } from '@syncfusion/ej2-base';
import Vue from 'vue';

var vueImport;
if (!isExecute || parseInt(allVue.version) < 3) {
    vueImport = Vue;
}
else {
    vueImport = Vue;
}
var SpeedDialItemsDirective = /** @class */ (function (_super) {
    __extends(SpeedDialItemsDirective, _super);
    function SpeedDialItemsDirective() {
        return _super.call(this, arguments) || this;
    }
    SpeedDialItemsDirective.prototype.render = function (createElement) {
        if (!isExecute) {
            var h = !isExecute ? gh : createElement;
            var slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = !isExecute ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    };
    SpeedDialItemsDirective.prototype.updated = function () {
        if (!isExecute && this.custom) {
            this.custom();
        }
    };
    SpeedDialItemsDirective.prototype.getTag = function () {
        return 'e-speeddial-items';
    };
    SpeedDialItemsDirective = __decorate([
        EJComponentDecorator({}, isExecute)
        
    ], SpeedDialItemsDirective);
    return SpeedDialItemsDirective;
}(vueImport));
export { SpeedDialItemsDirective };
export var SpeedDialItemsPlugin = {
    name: 'e-speeddial-items',
    install: function (Vue) {
        Vue.component(SpeedDialItemsPlugin.name, SpeedDialItemsDirective);
    }
};
/**
 * 'e-speeddialitem' directive represent a item of Vue SpeedDial
 * It must be contained in a SpeedDial component(`ejs-speeddial`).
 * ```html
 * <ejs-speeddial>
 *   <e-speeddialitems>
 *    <e-speeddialitem text='Cut'></e-speeddialitem>
 *    <e-speeddialitem text='Copy'></e-speeddialitem>
 *   </e-speeddialitems>
 * </ejs-speeddial>
 * ```
 */
var SpeedDialItemDirective = /** @class */ (function (_super) {
    __extends(SpeedDialItemDirective, _super);
    function SpeedDialItemDirective() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    SpeedDialItemDirective.prototype.render = function () {
        return;
    };
    SpeedDialItemDirective.prototype.getTag = function () {
        return 'e-speeddial-item';
    };
    SpeedDialItemDirective = __decorate([
        EJComponentDecorator({}, isExecute)
    ], SpeedDialItemDirective);
    return SpeedDialItemDirective;
}(vueImport));
export { SpeedDialItemDirective };
export var SpeedDialItemPlugin = {
    name: 'e-speeddial-item',
    install: function (Vue) {
        Vue.component(SpeedDialItemPlugin.name, SpeedDialItemDirective);
    }
};
